packages/datadog_flutter_plugin/lib/src/logs/ddlog_event.dart

```diff
@@ -0,0 +1,143 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-Present Datadog, Inc.
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'ddlog_event.g.dart';
+
+const commonJsonOptions = JsonSerializable(
+  anyMap: true,
+  includeIfNull: false,
+  explicitToJson: true,
+);
+
+Map<String, Object?> _attributesFromJson(Map? attributes) {
+  return attributes?.map(
+          (Object? key, Object? value) => MapEntry(key as String, value)) ??
+      {};
+}
+
+enum LogStatus {
+  debug,
+  info,
+  notice,
+  warn,
+  error,
+  critical,
+  emergency,
+}
+
+@commonJsonOptions
+class LogDevice {
+  final String architecture;
+
+  LogDevice({
+    required this.architecture,
+  });
+
+  factory LogDevice.fromJson(Map<String, dynamic> json) =>
+      _$LogDeviceFromJson(json);
+  Map<String, dynamic> toJson() => _$LogDeviceToJson(this);
+}
+
+@commonJsonOptions
+class LogEventDd {
```

Comment:
if we end new properties on the native side to `LogEvent`, will things still work? Or will it be the crash? Also the question comes how to we keep the two in sync.